test/e2e/broker_trigger_test.go

```diff
@@ -0,0 +1,250 @@
+	}
+
+	logger.Info("Subscriber pods running")
+
+	logger.Info("Creating Subscriber services")
+
+	for i, event := range eventsToReceive {
+		subscriberSvcName := name("svc", event.Type, event.Source)
+		service := test.Service(subscriberSvcName, ns, selectors[i])
+		if err := CreateService(clients, service, logger, cleaner); err != nil {
+			t.Fatalf("Error creating subscriber service: %v", err)
+		}
+	}
+
+	logger.Info("Subscriber services created")
+
+	logger.Info("Creating Triggers")
+
+	for _, event := range eventsToReceive {
+		triggerName := name("trigger", event.Type, event.Source)
+		// subscriberName should be the same as the subscriberSvc from before.
+		subscriberName := name("svc", event.Type, event.Source)
+		trigger := test.Trigger(triggerName, ns, event.Type, event.Source, defaultBrokerName, subscriberName)
+		err := CreateTrigger(clients, trigger, logger, cleaner)
+		if err != nil {
+			t.Fatalf("Error creating trigger: %v", err)
+		}
+	}
+
+	logger.Info("Triggers created")
+
+	logger.Info("Waiting for triggers to become ready")
+
+	// Wait for all of the triggers in the namespace to be ready.
+	if err := WaitForAllTriggersReady(clients, logger, ns); err != nil {
+		t.Fatalf("Error waiting for triggers to become ready: %v", err)
+	}
+
+	logger.Info("Triggers ready")
+
+	// These are the event types and sources that will be send.
+	eventsToSend := []test.TypeAndSource{
+		{eventType1, eventSource1},
+		{eventType1, eventSource2},
+		{eventType2, eventSource1},
+		{eventType2, eventSource2},
+	}
+
+	// We notice some crashLoopBacks in the filter and ingress pod creation.
+	// We then delay the creation of the sender pods in order not to miss events.
+	// TODO improve this
+	logger.Info("Waiting for filter and ingress pods to become running")
+	time.Sleep(waitForFilterPodRunning)
+
+	logger.Info("Creating event sender pods")
+
+	// Map to save the expected events per dumper so that we can verify the delivery.
+	expectedEvents := make(map[string][]string)
+	for _, eventToSend := range eventsToSend {
+		// Create cloud event.
+		// Using event type and source as part of the body for easier debugging.
+		body := fmt.Sprintf("Body-%s-%s", eventToSend.Type, eventToSend.Source)
+		cloudEvent := test.CloudEvent{
+			Source: eventToSend.Source,
+			Type:   eventToSend.Type,
+			Data:   fmt.Sprintf(`{"msg":%q}`, body),
+		}
+		// Create sender pod.
+		senderPodName := name("sender", eventToSend.Type, eventToSend.Source)
+		senderPod := test.EventSenderPod(senderPodName, ns, defaultBrokerUrl, cloudEvent)
+		if err := CreatePod(clients, senderPod, logger, cleaner); err != nil {
+			t.Fatalf("Error creating event sender pod: %v", err)
+		}
+
+		// Check on every dumper whether we should expect this event or not, and add its body
+		// to the expectedEvents map if so.
+		for _, eventToReceive := range eventsToReceive {
+			if shouldExpectEvent(&eventToSend, &eventToReceive, logger) {
```

Comment:
Is there anything testing that events do _not_ show up where they aren't supposed to?

Reply:
Good point. Added that as well!
Thanks for the comment